File: AppointmentSchedulerAPI/layers/BusinessLogicLayer/Model/ServiceOffer.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types;

namespace AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model
{
    public class ServiceOffer
    {
        public Assistant? Assistant { get; set; }
        public Service? Service { get; set; }
        public int? Id { get; set; }
        public Guid? Uuid { get; set; }
        public ServiceOfferStatusType? Status { get; set; }
        public List<ScheduledService>? ScheduledServices { get; set; }
    }
}

File: AppointmentSchedulerWeb/src/view-model/business-entities/service-offer.ts
import { ServiceOfferStatusType } from "./types/service-offer-status.types";
import { ServiceStatusType } from "./types/service-status.types";

export class ServiceOffer {
  name: string;
  price: number;
  minutes: number;
  description: string;
  uuid: string;
  status: ServiceOfferStatusType;
  assistant?: { name: string, uuid: string };
  serviceUuid?: string;
  serviceStatus?: ServiceStatusType;

  constructor(name: string, price: number, minutes: number, description: string, uuid: string, status: ServiceOfferStatusType, assistant?: { name: string, uuid: string }, serviceUuid?: string, serviceStatus?: ServiceStatusType) {
    this.name = name;
    this.price = price;
    this.minutes = minutes;
    this.description = description;
    this.uuid = uuid;
    this.status = status;
    this.assistant = assistant;
    this.serviceUuid = serviceUuid;
    this.serviceStatus = serviceStatus;
  }
}
